File: src/main/java/santiagoAndFerdy/vgs/model/RmiServer.java
package santiagoAndFerdy.vgs.model;

import java.rmi.RemoteException;
import java.rmi.registry.LocateRegistry;

/**
 * Created by Fydio on 3/7/16.
 */
public class RmiServer {
    private int port;
    private RmiServer rmiServer;

    public RmiServer(int port) throws RemoteException {
        this.port = port;

        LocateRegistry.createRegistry(port);
    }

    public int getPort() {
        return port;
    }
}

File: src/main/scala/grid/rmi/RmiServer.scala
package grid.rmi

import java.rmi.Naming
import java.rmi.registry._

import grid.cluster.Addressable

import scala.util.Try

/**
  * Created by Fydio on 4/13/16.
  */
object RmiServer {

  val rmiPort = 1099
  var registery: Registry = Try {
    LocateRegistry.createRegistry(rmiPort)
  }.toOption.getOrElse(LocateRegistry.getRegistry(rmiPort))

  def register(entity: Addressable): Unit = {
    val bindResult = Naming.rebind(entity.url, entity)
    println(s"Bound: $bindResult")
  }

}
